airflow/providers/docker/operators/docker.py

```diff
@@ -302,12 +310,12 @@ def _run_image_with_mounts(self, target_mounts, add_tmp_variable: bool) -> Optio
                     'stderr': True,
                     'stream': True,
                 }
-
-                return (
-                    self.cli.logs(**log_parameters)
-                    if self.xcom_all
-                    else self.cli.logs(**log_parameters, tail=1)
-                )
+                if self.xcom_all:
+                    return [stringify(line).strip() for line in self.cli.logs(**log_parameters)]
+                else:
+                    lines = [stringify(line).strip() for line in self.cli.logs(**log_parameters, tail=1)]
```

Comment:
```suggestion

                lines = [stringify(line).strip() for line in self.cli.logs(**log_parameters, tail=1)]
```

nit: we short circuit anyways, and I think this is easier to grok.

Reply:
I actually changed a bit - because I realized we had no case for "no lines returned" and well - obviouly - it failed so I had to fix it and reworked that one a bit.